### DISCUSSION TITLE
TailwindCSS with PostCSS module on generates scoped locals on Tailwind's utility classes

### DISCUSSION THREAD

**Discussion by papageno** [parcel-bundler/parcel#5369] [Q&A]
Here is a repo I created.

https://github.com/papegoje/parcel-not-playing-well-with-tailwindcss-and-css-modules.git

Here are the versions I am using.

```
package.json

...
    "parcel": "^2.0.0-nightly.447",
...
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "react-use": "^15.3.4",
...
    "autoprefixer": "^10.0.2",
    "postcss": "^8.1.7",
    "postcss-modules": "^3.2.2",
    "postcss-nested": "^5.0.1",
...
    "tailwindcss": "^1.9.6"
```

PostCSS config looks like this.

```
.postcssrc

{
  "modules": true,
  "plugins": {
    "tailwindcss": {},
    "postcss-nested": {},
    "autoprefixer": {},
  }
}
```

I also create a CSS for globals and import the Tailwind directives at its beginning.

```
global.css

@tailwind base;
@tailwind components;
@tailwind utilities;

...

```

I imported the global CSS to my main.js which is a React JSX to be imported to `index.html` using `<script src="./main.js">` tag

```
main.js

import React from "react"
import ReactDom from "react-dom"

import "./global.css"

const App = () => { ... }

ReactDom.render(<App />, document.getElementById("app"))
```

When I use Parce to build, I found in the built `index.[hash].css` that Tailwind's utility classes are like:

```
index.[hash].css

...

._mx-1_8acb9 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

._my-2_8acb9 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

._mx-2_8acb9 {
  margin-left: 0.5rem;
  margin-right: 0.5rem;
}

...

```

This totally loses its meaning to use those utility classes.

Am I doing something wrong in the PostCSS config?

**Comment by adityavm** [comment]
@papageno this configuration update worked for me https://github.com/parcel-bundler/parcel/issues/1930#issuecomment-488316106

make sure the path to the file is correct (if it's nested in sub-folders, it should be the full relative path)
